engine/src/main/java/org/terasology/config/flexible/FlexibleConfig.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import org.terasology.engine.SimpleUri;
+
+/**
+ * Stores multiple potentially related {@link Setting} instances that can be retrieved using their id.
+ */
+public interface FlexibleConfig {
+    /**
+     * Adds a {@link Setting} to the config if the id of the {@link Setting} is valid.
+     * @param setting The {@link Setting} to add.
+     * @return True if the {@link Setting} was added, false otherwise.
```

Comment:
Since consumers will be working with the interface itself, I prefer "warnings _will_ be issued" rather than "warnings _should_ be issued by an implementation" as it is more clear to the consumer and does not leave him wondering that the specific implementation of `FlexibleConfig` he refers to issues warnings or not. This way, all (good) implementations of `FlexibleConfig` will _have_ to issue warnings to comply with the base interface documentation.